test/RadworksDripsGovernance.t.sol

```diff
@@ -0,0 +1,126 @@
+
+    // Ensure proposal is in the expected state
+    IGovernor.ProposalState _state = governorBravo.state(_newProposalId);
+    assertEq(_state, IGovernor.ProposalState.Pending);
+
+    _jumpToActiveProposal(_newProposalId);
+
+    _delegatesCastVoteOnBravoGovernor(_newProposalId, FOR);
+    _jumpToVotingComplete(_newProposalId);
+
+    // Ensure the proposal has succeeded
+    _state = governorBravo.state(_newProposalId);
+    assertEq(_state, IGovernor.ProposalState.Succeeded);
+
+    // Queue the proposal
+    governorBravo.queue(_targets, _values, _calldatas, keccak256(bytes(_description)));
+
+    // Ensure the proposal is queued
+    _state = governorBravo.state(_newProposalId);
+    assertEq(_state, IGovernor.ProposalState.Queued);
+
+    _jumpPastProposalEta(_newProposalId);
+
+    // Execute the proposal
+    governorBravo.execute(_targets, _values, _calldatas, keccak256(bytes(_description)));
+
+    // Ensure the proposal is executed
+    _state = governorBravo.state(_newProposalId);
+    assertEq(_state, IGovernor.ProposalState.Executed);
+  }
+
+  function _grantNewPauserViaGovernance(address _newPauser) internal {
+    _proposePassAndExecuteDripsProposal(
+      "Grant Pauser role to an address",
+      _buildProposalData("grantPauser(address)", abi.encode(_newPauser))
+    );
+  }
+
+  function _revokePauserViaGovernance(address _newPauser) internal {
+    _proposePassAndExecuteDripsProposal(
+      "Revoke Pauser role from an address",
+      _buildProposalData("revokePauser(address)", abi.encode(_newPauser))
+    );
+  }
+
+  function testFuzz_grantPauserOnDrips(address _newPauser) public {
+    assummeNotTimelock(_newPauser);
+    address[] memory _originalPausers = drips.allPausers();
+
+    _grantNewPauserViaGovernance(_newPauser);
+
+    // Ensure the new pauser has been granted pauser role
+    assertEq(drips.isPauser(_newPauser), true);
+
+    // Ensure the the list of pausers got longer by 1
+    assertEq(_originalPausers.length + 1, drips.allPausers().length);
+
+    // Ensure the new pauser can pause the DRIPS protocol
+    vm.prank(_newPauser);
+    drips.pause();
+    assertTrue(drips.isPaused());
+
+    // Ensure the new pauser can un-pause the DRIPS protocol
+    vm.prank(_newPauser);
+    drips.unpause();
+    assertFalse(drips.isPaused());
+  }
+
+  function testFuzz_revokePauserOnDrips(address _newPauser) public {
+    assummeNotTimelock(_newPauser);
+    _grantNewPauserViaGovernance(_newPauser);
+
+    // Ensure the new pauser has been granted pauser role
+    assertEq(drips.isPauser(_newPauser), true);
+
+    _revokePauserViaGovernance(_newPauser);
+
+    // Ensure the new pauser has subsequently had pauser role revoked
+    assertEq(drips.isPauser(_newPauser), false);
+
+    // Ensure the newly-revoked pauser cannot pause the DRIPS protocol
+    vm.prank(_newPauser);
+    vm.expectRevert("Caller not the admin or a pauser");
+    drips.pause();
```

Comment:
Nit: some of these assertions seem like they could be broken out into separate tests. It could allow us to have more descriptive test names for specific situations. For example, in this last assert we are checking a revert, and to follow the foundry test best practices we would structure our test name to reflect this with a name like `testFuzz_RevertIf_CallerIsNotAPauserOrAdmin`.

Reply:
Also, it would be good to have tests with an existing pauser and admin to be more explicit about other edge cases.

Reply:
Done: d5ab4e06fd3aac54151da3b53cb244e77d0a3cfc